15 use one config

edestecd opened the pull request:
- [x] Spec Tests for lastpass::config
- [x] Doc in README about lastpass::config

edestecd committed:
Fixes #15 - Use one config file

edestecd committed:
Provide reasonable defaults for group and config_dir

No need to validate these if we have defaults. Configs are only generated if set

edestecd committed:
Fix file read in lastpass_functions

edestecd committed:
Add more spec tests

edestecd committed:
Document lastpass::config in the README

dstepe commented on README.md:
```diff
@@ -51,21 +52,42 @@ class { 'lastpass': }
+The module provides two helper scripts to facilitate automated login for unattended systems. These scripts require the LastPass username and password be stored in $LASTPASS_HOME/login. The module will create this if the username and password parameters are provided, or this can be created manually for the desired user. The default value of $LASTPASS_HOME is '$HOME/.lpass'.
 
-The lpasslogin script will read the $LASTPASS_HOME/user file and run 'lpass login $USERNAME'. If $LASTPASS_HOME/pw is present, the profile script installed by the module will set $LASTPASS_ASKPASS to the lpasspw script, which reads $LASTPASS_HOME/pw and prints it to STDOUT in response to the CLI.
+The lpasslogin script will read the $LASTPASS_HOME/login file and run 'lpass login $LPASS_USERNAME'. If $LASTPASS_HOME/login is present, the profile script installed by the module will set $LASTPASS_ASKPASS to the lpasspw script, which reads $LASTPASS_HOME/login and prints $LASTPASS_PASSWORD it to STDOUT in response to the CLI.
```
The variable names should LPASS_HOME, LPASS_USERNAME and LPASS_ASKPASS (I say after spending 30 minutes poking at this before reading the reference).

edestecd replied:
fixed. Also added an example of the $LPASS_HOME/login file and fixed several other env var mistakes.

dstepe commented:
I've been able to get both interactive and non-interactive working through manual configuration. Setting up non-interactive login requires adding the correct LPASS_ASKPASS variables to the $LPASS_HOME/env file and the correct variables to $LPASS_HOME/login file. I had hoped that we could always have LPASS_ASKPASS set and let it decide how to handle the password based on the login file. That's the intent of the conditional in the lpasspw script, but is moot since LPASS_ASKPASS is not set automatically.

I think the docs need to be clearer about setting up the non-interactive manually. It took me a few tries. I can try and work on this at some point.

It's also very import to test the non-interactive login before running puppet. If the login ends up prompting for a password, the puppet run will hang.

edestecd committed:
Fix errors in README

edestecd commented:
We could make LPASS_ASKPASS an option to the lastpass class, if you want it on by default.
I kinda thought it was nice to not make any config file, unless the end user asked by passing a param, but I can see the advantage of the automated login just working out of the box by plugging in the username and password either manually or through the module.

dstepe commented:
I'm not sure the use case will ever come up. The most likely case is a single user (probably root) being configured through options. Maybe the password is set manually. The other use case is to just install the CLI via puppet and then the sys admin provides instructions to users who wish to configure lpass, in which case the behavior without setting LPASS_ASKPASS is actually the better choice. I'm content with just a comment in the docs about setting that manually in the env file if necessary.

edestecd committed:
More doc about LPASS_ASKPASS

edestecd commented:
@dstepe More doc about LPASS_ASKPASS added

tepeds committed:
Make the status check less brittle

The status response started including the newline for some reason. Changing from an exact string match to a suitable regex.

tepeds committed:
Merge branch '15-use-one-config' of github.com:MiamiOH/puppet-lastpass into 15-use-one-config
